src/main/java/nl/tudelft/dnainator/ui/models/GraphItem.java

```diff
@@ -6,81 +6,157 @@
+		getChildren().add(content);
+		getChildren().add(childContent);
+		loadContent();
+	}
+
+	/**
+	 * Check whether this object intersects with the given viewport bounds.
+	 * @param bounds	The given viewport bounds.
+	 * @return	True if (partially) in viewport, false otherwise.
+	 */
+	public boolean isInViewport(Bounds bounds) {
+		return bounds.contains(content.localToParent(0, 0));
+	}
+
+	/*
+	 * Load the drawable content of the graph itself.
+	 */
+	private void loadContent() {
+		for (int i = 0; i < SLICES; i++) {
+			int width = NO_CLUSTERS * RANK_WIDTH / SLICES;
+			Rectangle rectangle = new Rectangle(width, RANK_WIDTH, Color.BLACK);
+			rectangle.setTranslateX(i * width);
+			content.getChildren().add(rectangle);
 		}
 	}
 
-	@Override
-	public Graph getGraph() {
-		return graph;
+	/*
+	 * Load the drawable content of the graph's children, given the bounds of the viewing port.
+	 */
+	private void loadChildren(Bounds bounds) {
+		int minRank = (int) (Math.max(bounds.getMinX() / RANK_WIDTH, 0));
+		int maxRank = (int) (bounds.getMaxX() / RANK_WIDTH);
+
+		System.out.println("load iteration: " + minRank + " -> " + maxRank);
+		List<String> roots = graph.getRank(minRank).stream()
+				.map(SequenceNode::getId).collect(Collectors.toList());
+		Map<Integer, List<Cluster>> result = graph.getAllClusters(roots, maxRank,
+				(int) (bounds.getWidth() / CLUSTER_DIVIDER));
+
+		clusters.clear();
+		childContent.getChildren().clear();
+
+		result.forEach(this::loadRank);
+		clusters.values().forEach(this::loadEdges);
+	}
+
+	/*
+	 * Load the drawable content of the edges for all displayed clusters.
+	 */
+	private void loadEdges(ClusterDrawable cluster) {
+		childContent.getChildren().addAll(
+				cluster.getClustered().stream().flatMap(e -> e.getOutgoing().stream())
+						.filter(clusters::containsKey)
+						.filter(i -> clusters.get(i) != cluster)
+						.map(o -> new DrawableEdge(cluster, clusters.get(o)))
+						.collect(Collectors.toList()));
 	}
 
-	@Override
-	public Map<String, ClusterItem> getClusters() {
-		return clusters;
+	private void loadRank(Integer key, List<Cluster> value) {
+		for (int i = 0; i < value.size(); i++) {
+			ClusterDrawable cluster = new ClusterDrawable(value.get(i).getNodes());
+			cluster.getClustered().forEach(e -> clusters.put(e.getId(), cluster));
+			cluster.setTranslateX(key * RANK_WIDTH);
+			cluster.setTranslateY(i * RANK_WIDTH - value.size() * RANK_WIDTH / 2);
+			childContent.getChildren().add(cluster);
+		}
 	}
 
-	@Override
-	public ModelItem getRoot() {
-		return this;
+	/**
+	 * Toggle between displaying own content or children.
+	 * @param bounds     The bounds of the viewport.
+	 * @param visible    True if the content needs to be visible, false otherwise.
+	 */
+	public void toggle(Bounds bounds, boolean visible) {
+		if (visible && !content.isVisible()) {
+			childContent.getChildren().clear();
+			content.setVisible(true);
+			loadContent();
+		}
+		if (!visible) {
```

Comment:
The content is not cleared on purpose.
Since the 'content' is only the rectangle for the graph, it does not really take up any memory and does not need to be cleared.